unime/src/lib/components/SettingsSwitch.svelte

```diff
@@ -0,0 +1,35 @@
+<script lang="ts">
+  import type { Snippet } from 'svelte';
+
+  import Switch from './Switch.svelte';
+
+  interface Props {
+    initialChecked?: boolean;
+    disabled?: boolean;
+    onchange: (checked: boolean) => void;
+    icon?: Snippet;
+    children: Snippet;
+  }
+
+  let { icon, initialChecked = false, disabled = false, onchange, children }: Props = $props();
+</script>
+
+<div
+  data-component="SettingsSwitch"
+  class="flex h-14 w-full items-center gap-4 rounded-xl bg-background-alt p-4 {disabled
+    ? 'pointer-events-none opacity-50'
+    : ''}"
+>
+  {#if icon}
+    <div class="flex items-center justify-center">
+      {@render icon()}
+    </div>
+  {/if}
+  <div class="grow">
+    <Switch {initialChecked} {onchange}>
+      <span class="text-sm font-medium text-text">
```

Comment:
In the previous `SettingsEntry`, we sticked closely to the designs and used `text-[13px]/[24px]` which differs slightly from `text-sm` (`text-[14px]/[20px]`). We can introduce the proposed change, but it could have a potentially breaking visual effect regarding the font size used in other pages and components (which still use `text-[13px]/[24px]`). I'm would also support to redefine the default `text-sm` at a later stage.